utils/text.go

```diff
@@ -8,7 +8,7 @@ import (
 var snakedChars = regexp.MustCompile(`[^\p{L}\d_]+`)
 
 // treats sequences of letters/numbers/_/' as tokens, and symbols as individual tokens
-var wordTokenRegex = regexp.MustCompile(`[\pL\pN_']+|\pS`)
+var wordTokenRegex = regexp.MustCompile(`[\pM\pL\pN_']+|\pS`)
```

Comment:
FYI this includes...

https://www.fileformat.info/info/unicode/category/Mn/list.htm
https://www.fileformat.info/info/unicode/category/Mc/list.htm
https://www.fileformat.info/info/unicode/category/Me/list.htm

There's some weird stuff in `Me`.. but maybe for simplicity best to just match all of `M` like you have it above.